src/vector/app.tsx

```diff
@@ -38,6 +38,8 @@ import { createClient } from "matrix-js-sdk/src/matrix";
 
 let lastLocationHashSet: string = null;
 
+console.log(`Application is running in ${process.env.NODE_ENV} mode`);
+
```

Comment:
I'm not entirely sure of the information we're trying to get out of this log line
The application is not running in a node environment, so it is unlikely that the environment variable will be set.

What is more important though is for this variable to be set at build time so that webpack can work its magic and pick the correct React build

Reply:
@gsouquet `process.env.NODE_ENV` will be replaced by webpack on compilation time, that's how it's even done in React, so the prod mode is just a few `if` statements with `if (process.env.NODE_ENV === 'development')` 😄

Reply:
It literally just uses https://webpack.js.org/plugins/define-plugin/ to do hard replacements 'global' variables like these.

Reply:
Right, probs worth adding a little comment explaining what it does, as it is a bit unclear to see a `NODE_ENV` reference for an app that does not require node at runtime.

Reply:
That's a very common pattern though, not sure where I should add the information about it, as we're already using it in js-sdk AFAIK.

Reply:
What is the advantage of using the replacing behaviour if this code doesn't ever run in node?

Reply:
https://webpack.js.org/guides/production/#specify-the-mode
I'll just leave a link to the docs, will do much better at explaining the job than me. 
The replace behavior is for making things macro-like, due to how dead code removal on minifying step works.
If you have some kind of error reporting for dev only, you can add `if  (process.env.NODE_ENV === 'development') throw new Error('This should never happen, check this out')`, that would get happily removed on production build without ever going to break any user's flow due to code replacement, as in prod build it'd get replaced into
` if (false) throw new Error('This should never happen, check this out')`  and gets happily removed out during minification due to dead code removal.

Reply:
ah cool, thanks